crypto/merkletrie/node.go

```diff
@@ -183,8 +181,8 @@ func (n *node) add(cache *merkleTrieCache, d []byte, path []byte) (nodeID stored
 			for i, child := range n.children {
 				if d[0] < child.hashIndex {
 					pnode.children[i] = childEntry{
-						hashIndex: d[0],
 						id:        childNodeID,
+						hashIndex: d[0],
```

Comment:
FYI: This is because we write them in this order in all the nearby initializations.